onnx/backend/test/runner/__init__.py

```diff
@@ -169,15 +169,19 @@ def tests(self):  # type: () -> Type[unittest.TestCase]
     def assert_similar_outputs(cls, ref_outputs, outputs, rtol, atol):  # type: (Sequence[Any], Sequence[Any], float, float) -> None
         np.testing.assert_equal(len(outputs), len(ref_outputs))
         for i in range(len(outputs)):
-            np.testing.assert_equal(outputs[i].dtype, ref_outputs[i].dtype)
-            if ref_outputs[i].dtype == np.object:
-                np.testing.assert_array_equal(outputs[i], ref_outputs[i])
+            if isinstance(outputs[i], (list, tuple)):
+                for j in range(len(outputs[i])):
+                    cls.assert_similar_outputs(ref_outputs[i][j], outputs[i][j], rtol, atol)
             else:
-                np.testing.assert_allclose(
-                    outputs[i],
-                    ref_outputs[i],
-                    rtol=rtol,
-                    atol=atol)
+                np.testing.assert_equal(outputs[i].dtype, ref_outputs[i].dtype)
+                if ref_outputs[i].dtype == np.object:
```

Comment:
It would be nice to have a comment to explain what case is handled by the then branch. (I understand that this is a problem with the pre-existing code, but it would help readers/reviewers/maintainers in the future.)

Reply:
In case you know.